js/src/util/wallet.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import ethUtil from 'ethereumjs-util';
+import scrypt from 'scryptsy';
+import Transaction from 'ethereumjs-tx';
+import crypto from 'crypto';
+import aes from 'browserify-aes';
+
+// Adapted from https://github.com/kvhnuke/etherwallet/blob/mercury/app/scripts/myetherwallet.js
+
+export class Wallet {
+
+  static fromJson (json, password) {
+    if (json.version !== 3) {
+      throw new Error('Only V3 wallets are supported');
+    }
+
+    const { kdf } = json.crypto;
+    const kdfparams = json.crypto.kdfparams || {};
+    const pwd = new Buffer(password);
```

Comment:
[`new Buffer` is deprecated, used `Butter.from`.](https://nodejs.org/api/buffer.html#buffer_new_buffer_string_encoding)